### ISSUE TITLE
Packaging gopkg.in/ozeidan/fuzzy-patricia.v3 [#1053520]

### ISSUE THREAD

**Issue by jmkim** [jesseduffield/lazygit-debian#40]
https://salsa.debian.org/go-team/packages/golang-github-ozeidan-fuzzy-patricia

**Comment by WeepingClown13** [comment]
Hi, can I work on this?

**Comment by jmkim** [comment]
@WeepingClown13 welcome \o/ Go ahead

**Comment by WeepingClown13** [comment]
Hi, I have some updates. Been a while since I packaged this and sent an RFS (I wanted to update the labels but seems you need repo access for that), but I got some feedback only now. You can read about it [here](https://lists.debian.org/debian-go/2023/10/msg00047.html).
@jesseduffield what do you think about the things asked in the feedback?

**Comment by jmkim** [comment]
(for logging the infos)
ITP : https://bugs.debian.org/1053520

**Comment by Maytha8** [comment]
I agree with Nilesh's POV on this one, especially since [lazygit is the only package that depends on ozeidan/fuzzy-patricia](https://github.com/ozeidan/fuzzy-patricia/network/dependents) AFAICS.

~~@WeepingClown13 Consider patching ozeidan's commits on top of tchap/go-patricia.
For example: [here](https://salsa.debian.org/go-team/packages/golang-github-protonmail-gluon/-/blob/debian/sid/debian/rules?ref_type=heads#L8-17), where I've patched the commits from ProtonMail/go-mbox (a fork) on top of the existing Debian package for emersion/go-mbox.~~

**Comment by WeepingClown13** [comment]
@Maytha8 yes, I also agree with what Nilesh said, I just wanted to confirm what Jesse thinks so that if possible this whole scenario can be avoided.

And thank you for the patch reference, that helps. Although I believe this will have to be done in lazygit when lackaging it as I understand it. I'll look in to it nonetheless, and I appreciate the help :)

**Comment by Maytha8** [comment]
> Although I believe this will have to be done in lazygit when lackaging it as I understand it.

~~@jmkim What do you think?~~

Never mind, it's a really fragile implementation.

**Comment by Maytha8** [comment]
Finally uploaded to NEW!

**Comment by jmkim** [comment]
Someone know the status of this package? o/ @Maytha8 @WeepingClown13

I cannot find this in followings:
* https://incoming.debian.org/debian-buildd/pool/main/g/
* http://tracker.debian.org/pkg/golang-github-ozeidan-fuzzy-patricia

**Comment by WeepingClown13** [comment]
Rejected with:

"please also mention the go-patricia authors in your debian/copyright."

I can fix that up and upload again.

**Comment by jmkim** [comment]
Re uploaded by @WeepingClown13, many thanks for your work!

https://ftp-master.debian.org/new/golang-github-ozeidan-fuzzy-patricia_3.0.0-1.html

**Comment by jmkim** [comment]
golang-github-ozeidan-fuzzy-patricia ACCEPTED \o/

https://tracker.debian.org/pkg/golang-github-ozeidan-fuzzy-patricia

**Comment by jmkim** [comment]
Migration to testing BLOCKED:

* https://qa.debian.org/excuses.php?package=golang-github-ozeidan-fuzzy-patricia
* https://buildd.debian.org/status/logs.php?suite=sid&arch=all&pkg=golang-github-ozeidan-fuzzy-patricia&ver=3.0.0-2
* https://buildd.debian.org/status/fetch.php?pkg=golang-github-ozeidan-fuzzy-patricia&arch=all&ver=3.0.0-2&stamp=1744327673&raw=0

Reason: 

* Build failed on buildd during `dh_auto_test`:
    ```bash
    === RUN   TestParticiaTrie_DeleteLeakageSparse
        patricia_sparse_test.go:484: Size=0, Total=1, Trie state:
             <nil>
            
        patricia_sparse_test.go:485: Heap space leak, grew 4008 bytes (from 147800 to 151808)
    --- FAIL: TestParticiaTrie_DeleteLeakageSparse (0.03s)
    ```

Related issues:
* `TestTrie_DeleteLeakageDense` : https://github.com/tchap/go-patricia/issues/39
* `TestParticiaTrie_DeleteLeakageSparse` : https://github.com/tchap/go-patricia/issues/20

**Comment by jmkim** [comment]
Fixed and uploaded by @WeepingClown13 \o/
